enhancements/network/user-defined-network-segmentation.md

```diff
@@ -0,0 +1,947 @@
+with user-defined networks will still attach to the cluster default network with limited access to Kubernetes system resources.
+Pods with user-defined networks will have at least two network interfaces, one connected to the cluster default network and one
+connected to the user-defined network. Pods with user-defined networks will use the user-defined network as their default
+gateway.
+* Allow multiple namespaces per network.
+* Support cluster ingress/egress traffic for user-defined networks, including secondary networks.
+* Support for ingress/egress features where possible:
+    * EgressQoS
+    * EgressService
+    * EgressIP
+    * Load Balancer and NodePort Services, as well as services with External IPs.
+* In addition to ingress service support, there will be support for Kubernetes services in user-defined networks. The
+scope of reachability to that service as well as endpoints selected for that service will be confined to the network
+and corresponding namespace(s) where that service was created.
+* Support for pods to continue to have access to the cluster default primary network for DNS and KAPI service access.
+* Kubelet healthchecks/probes will still work on all pods.
+* OpenShift Router/Ingress will work with some limitations for user-defined networks.
+
+### Non-Goals
+
+* Allowing different service CIDRs to be used in different networks.
+* Localnet will not be supported initially for primary networks.
+* Allowing multiple primary networks per namespace.
+* Hybrid overlay support on user-defined networks.
+
+### Future-Goals
+
+* DNS lookup for pods returning records for IPs on the user-defined network. In the first phase DNS will return the pod
+IP on the cluster default network instead.
+* Admin ability to configure networks to have access to all services and/or expose services to be accessible from all
+networks.
+* Ability to advertise user-defined networks to external networks using BGP/EVPN. This will enable things like:
+    * External -> Pod ingress per VRF (Ingress directly to pod IP)
+    * Multiple External Gateway (MEG) in a BGP context, with ECMP routes
+* Allow connection of multiple networks via explicit router API configuration.
+
+## Proposal
+
+By default in OVN-Kubernetes, pods are attached to what is known as the “cluster default" network, which is a routed network
+divided up into a subnet per node. All pods will continue to have an attachment to this network, even when assigned a
+different primary network. Therefore, when a pod is assigned to a user-defined network, it will have two interfaces, one
+to the cluster default network, and one to the user-defined network. The cluster default network is required in order to provide:
+
+1. KAPI service access
+2. DNS service access
+3. Kubelet healthcheck probes to the pod
+4. User defined ports that will be exposed on the cluster default network. This may be used for things like prometheus
```

Comment:
do I need to create a separate spike card for this point perhaps?

Reply:
for sure

Reply:
https://issues.redhat.com/browse/SDN-5031